### PR TITLE
[DO NOT MERGE] ebpf: add webhook support

### PR DESCRIPTION
**Pull request by josedonizetti**
## Initial Checklist

This PR adds webhook support to `tracee-ebpf`, which is part of the task to have everything as an event, by merging `tracee-ebpf` and `tracee-rules`. The `--webhook` flag allows multiple values (currently limited) so we can extend it further later, also it is not a printer, because we should be able to print/send webhooks at the same time, plus webhook should evolve to much more complex than a printer later, eg:

- other formats (template? gob?)
- secret header value
- tls
- mutual tls (spiffe integration, etc)
- circuit break
- event filtering

## Description (git log)

```
commit 28ad426144834803a0093dcd67c0b57a34ca4d14 (HEAD -> add-webhook-tracee-ebpf, origin/add-webhook-tracee-ebpf)
Author: Jose Donizetti <jdbjunior@gmail.com>
Date:   Wed Nov 16 21:54:52 2022 -0300

    ebpf: add webhook support
```

Copy and paste the git log here.

Fixes: https://github.com/aquasecurity/tracee/issues/2362

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

Reproduce the test by running:

```
sudo tracee-ebpf --webhook url=http://localhost:8080 --webhook timeout=5s --webhook format=json --trace event=execve
```

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

File: pkg/webhook/webhook.go
Comment (geyslan) [inline pkg/webhook/webhook.go:20]: Will this method make changes in the receiver in the future? If not, I suppose it can be a value receiver.

File: pkg/webhook/webhook.go
Comment (geyslan) [inline pkg/webhook/webhook.go:54]: ditto

**Review by geyslan — APPROVED** [review]
Tested with `for ((i=0;i<5;i++)); do echo -e "\n\nnc INSTANCE $i\n"; nc -l -p 8080; done` and worked.

File: cmd/tracee-ebpf/main.go
Comment (josedonizetti) [reply cmd/tracee-ebpf/main.go:256]: :+1: I'll add a task to look into this, once we mege this PR. k?

**Comment by josedonizetti** [comment]
I've added a DO NOT MERGE to the PR title after speaking with @NDStrahilevitz. This PR is dangerous to the user because it does not filter which events go on the webhook. Tracee collects lots and lots of data, and the current data doesn't make sense to go down a webhook, webhooks intentions are not to take data out, for this other strategies should be used, GRPC streaming sever, fluentd, etc

This PR is part of the task `everything is an event`, which would bring signatures are events, and those events make sense to be sent down the webhook because they are supposed to generate alerts. Let's wait until others PRs for the everything is an event task is done, before merging this one.

File: cmd/tracee-ebpf/main.go
Comment (yanivagman) [inline cmd/tracee-ebpf/main.go:255]: Adding this here means that for every event we will create and send an http packet, is that right?
If so, it has very bad implications on performance which we should avoid.
Maybe as a first step we should simply make the webhook another printer? WDYT?

**Comment by NDStrahilevitz** [comment]
@josedonizetti can we move this to draft to prevent accidental merging until you have the other relevant infrastructure merged?

**Pull request closed without merging**
